### DISCUSSION TITLE
LIEF leads to undefined symbol for a system lib

### DISCUSSION THREAD

**Discussion by shubham4060** [lief-project/LIEF#999] [General]
Hi,

i have been recently working with LIEF to patch elf files. I came across this weird case. There is a system lib named libicuuc.so.72. using lief i just parse this lib, enumerate the symbols and write back(no modification to lib). and doing this leads to undefined error when loading a library that is dependent on this.

code i used to patch
```
for lib in os.listdir(lib_dir):
    if lib == 'libicuuc.so.72':
        lib_lief = lief.parse(f"{lib_dir}/{lib}")
        idxs = []
        for i,symbol in enumerate(lib_lief.symbols):
            if symbol.name in symb_set:
                idxs.append(i)
       lib_lief.write(f"{lib_dir}/{lib}")
```

before runninhg this everything works smoothly but post running this i am getting the below error
`OSError: /home/shubham/shubham_test/libicui18n.so.72: undefined symbol: _ZTIN6icu_7216ICULocaleServiceE`

all these system libs are present in the same directory as my script(in shubham_test)

a sample script that reproduced this error at my end

```
import ctypes,ctypes.util
lib_handle = ctypes.CDLL('./my_lib.so', mode=ctypes.RTLD_LOCAL)
```

where my_lib.so is a elf library i produced that is dependent on libicuuc.so

Also, this does not errors out if i say i try to load directly libicui18n.so.72.

can anyone please help as to what might be the issue or can point me as to where i should start looking to figure this out?

Thanks,

**Comment by romainthomas** [comment]
Duplicate: https://github.com/lief-project/LIEF/issues/983
